bson/_cbsonmodule.c

```diff
@@ -447,6 +447,26 @@ static long _type_marker(PyObject* object) {
     return type;
 }
 
+/* Fill out a type_registry_t* from a TypeRegistry object.
+ *
+ * Return 1 on success. options->document_class is a new reference.
+ * Return 0 on failure.
+ */
+int convert_type_registry(PyObject* registry_obj, type_registry_t* registry) {
+    registry->encoder_map = PyObject_GetAttrString(registry_obj, "_encoder_map");
+    if (registry->encoder_map == NULL)
+        return 0;
+
+    registry->decoder_map = PyObject_GetAttrString(registry_obj, "_decoder_map");
+    if (registry->decoder_map == NULL)
+        return 0;
```

Comment:
Before returning `0` here we need to decref the `registry->encoder_map` object.

Reply:
I suggest writing this function in the style where we:
- declare all `PyObjects *` as NULL upfront.
- have a single error label that xdecrefs all objects
- on any error: `goto <errorlabel>`